[DBG_X] DTMuonMillepede: array subscript ... is outside outside array bounds

In the recent [DBG_X IB](https://cmssdt.cern.ch/SDT/cgi-bin/buildlogs/el8_amd64_gcc12/CMSSW_13_3_DBG_X_2023-10-26-2300/Alignment/MuonAlignmentAlgorithms), the compiler emits warnings about accessing array elements outside it's bounds:
```
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc: In member function 'getCqcMatrix':
  .../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:582:44: warning: array subscript 5 is outside array bounds of 'void[40]' [-Warray-bounds]
   582 |             Eta[counterLayer][counterLayer2] = -1.0 / (4.0);
      |                                            ^
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:574:39: note: at offset 40 into object of size 40 allocated by 'operator new []'
  574 |       Eta[counterLayer] = new double[5];
      |                                       ^
  .../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:582:44: warning: array subscript 5 is outside array bounds of 'void[40]' [-Warray-bounds]
   582 |             Eta[counterLayer][counterLayer2] = -1.0 / (4.0);
      |                                            ^
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:574:39: note: at offset 40 into object of size 40 allocated by 'operator new []'
  574 |       Eta[counterLayer] = new double[5];
      |                                       ^
  .../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:582:44: warning: array subscript 6 is outside array bounds of 'void[40]' [-Warray-bounds]
   582 |             Eta[counterLayer][counterLayer2] = -1.0 / (4.0);
      |                                            ^
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:574:39: note: at offset 48 into object of size 40 allocated by 'operator new []'
  574 |       Eta[counterLayer] = new double[5];
      |                                       ^
  .../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:580:44: warning: array subscript 5 is outside array bounds of 'void[40]' [-Warray-bounds]
   580 |             Eta[counterLayer][counterLayer2] = 3.0 / (4.0);
      |                                            ^
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:574:39: note: at offset 40 into object of size 40 allocated by 'operator new []'
  574 |       Eta[counterLayer] = new double[5];
      |                                       ^
  .../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:582:44: warning: array subscript 6 is outside array bounds of 'void[40]' [-Warray-bounds]
   582 |             Eta[counterLayer][counterLayer2] = -1.0 / (4.0);
      |                                            ^
.../src/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc:574:39: note: at offset 48 into object of size 40 allocated by 'operator new []'
  574 |       Eta[counterLayer] = new double[5];
      |                                       ^
(... etc ...)
```
(54 warnings total)

I think these warnings are valid: the `Eta` array is defined as `double[12][5]`:
https://github.com/cms-sw/cmssw/blob/a05e5a462c054b985d6bd9e9d85b02e725f3ee1e/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc#L570

https://github.com/cms-sw/cmssw/blob/a05e5a462c054b985d6bd9e9d85b02e725f3ee1e/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc#L574

however the `counterLayer2` index goes from 0 to 11 - https://github.com/cms-sw/cmssw/blob/a05e5a462c054b985d6bd9e9d85b02e725f3ee1e/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc#L575

So we should either change Eta to be 11x11, or change the `counterLayer2` loop.

--- a/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc
+++ b/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc
@@ -668,10 +668,10 @@ TMatrixD DTMuonMillepede::getbqcMatrix(int wh, int st, int se) {
       }
     }
     std::array<std::array<double, 12>, 12> Eta{};
-    for (size_t counterLayer = 0; counterLayer < 12; counterLayer++) {
+    for (size_t counterLayer = 0; counterLayer < Eta.size(); counterLayer++) {
       if (counterLayer > 7 && st == 4)
         continue;
-      for (size_t counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
+      for (size_t counterLayer2 = 0; counterLayer2 < Eta[counterLayer].size(); counterLayer2++) {
         if (counterLayer2 > 7 && st == 4)
           continue;
         if ((counterLayer2 < 4 && counterLayer < 4) || (counterLayer2 > 3 && counterLayer > 3)) {

--- a/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc
+++ b/Alignment/MuonAlignmentAlgorithms/src/DTMuonMillepede.cc
@@ -1,5 +1,6 @@
 #include "Alignment/MuonAlignmentAlgorithms/interface/DTMuonMillepede.h"
 
+#include <array>
 #include <iostream>
 
 DTMuonMillepede::DTMuonMillepede(std::string path,
@@ -567,13 +568,13 @@ TMatrixD DTMuonMillepede::getCqcMatrix(int wh, int st, int se) {
         }
       }
     }
-    double **Eta = new double *[12];
-    for (int counterLayer = 0; counterLayer < 12; counterLayer++) {
+
+    std::array<std::array<double, 12>, 12> Eta{};
+    for (size_t counterLayer = 0; counterLayer < Eta.size(); counterLayer++) {
       if (counterLayer > 7 && st == 4)
         continue;
-      Eta[counterLayer] = new double[5];
-      for (int counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
-        if (counterLayer > 7 && st == 4)
+      for (size_t counterLayer2 = 0; counterLayer2 < Eta[counterLayer].size(); counterLayer2++) {
+        if (counterLayer2 > 7 && st == 4)
           continue;
         if ((counterLayer2 < 4 && counterLayer < 4) || (counterLayer2 > 3 && counterLayer > 3)) {
           if (counterLayer == counterLayer2) {
@@ -581,20 +582,18 @@ TMatrixD DTMuonMillepede::getCqcMatrix(int wh, int st, int se) {
           } else {
             Eta[counterLayer][counterLayer2] = -1.0 / (4.0);
           }
-        } else {
-          Eta[counterLayer][counterLayer2] = 0.0;
         }
       }
     }
 
     for (int counterDeg = 0; counterDeg < 5; counterDeg++) {
-      for (int counterLayer = 0; counterLayer < 12; counterLayer++) {
+      for (size_t counterLayer = 0; counterLayer < 12; counterLayer++) {
         if (counterLayer > 7 && st == 4)
           continue;
-        for (int counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
+        for (size_t counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
           if (counterLayer2 > 7 && st == 4)
             continue;
-          for (int counterLayer3 = 0; counterLayer3 < 12; counterLayer3++) {
+          for (size_t counterLayer3 = 0; counterLayer3 < 12; counterLayer3++) {
             if (counterLayer3 > 7 && st == 4)
               continue;
             matrix(5 * counterLayer2 + counterDeg, 5 * counterLayer3 + counterDeg) +=
@@ -668,13 +667,12 @@ TMatrixD DTMuonMillepede::getbqcMatrix(int wh, int st, int se) {
         }
       }
     }
-    double **Eta = new double *[12];
-    for (int counterLayer = 0; counterLayer < 12; counterLayer++) {
+    std::array<std::array<double, 12>, 12> Eta{};
+    for (size_t counterLayer = 0; counterLayer < 12; counterLayer++) {
       if (counterLayer > 7 && st == 4)
         continue;
-      Eta[counterLayer] = new double[5];
-      for (int counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
-        if (counterLayer > 7 && st == 4)
+      for (size_t counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
+        if (counterLayer2 > 7 && st == 4)
           continue;
         if ((counterLayer2 < 4 && counterLayer < 4) || (counterLayer2 > 3 && counterLayer > 3)) {
           if (counterLayer == counterLayer2) {
@@ -689,10 +687,10 @@ TMatrixD DTMuonMillepede::getbqcMatrix(int wh, int st, int se) {
     }
 
     for (int counterDeg = 0; counterDeg < 5; counterDeg++) {
-      for (int counterLayer = 0; counterLayer < 12; counterLayer++) {
+      for (size_t counterLayer = 0; counterLayer < 12; counterLayer++) {
         if (counterLayer > 7 && st == 4)
           continue;
-        for (int counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
+        for (size_t counterLayer2 = 0; counterLayer2 < 12; counterLayer2++) {
           if (counterLayer2 > 7 && st == 4)
             continue;
           float mean = 0;

